### ISSUE TITLE
Incorrect outline when h1 appears after nav header

### ISSUE THREAD

**Issue by TheDutchCoder** [validator/validator#535]
In the following structure, the outline is incorrectly reported:

```
  <body>

    <header>
      <nav>
        <h2>Site navigation</h2>
      </nav>
    </header>

    <h1>Main heading</h1>

    <footer>Site Footer</footer>

  </body>
```

The validator reports:
![image](https://user-images.githubusercontent.com/1668207/28077135-d87c8458-662e-11e7-927a-dfb4380a000e.png)

The validator should hoist the h1 as being the document's heading and use the h2 as a second level heading under it. Otherwise users would be forced to include an h1 before the nav, which would e.g. default to the website's name or something similar, which is bad for both the general outline, as well as SEO.

The following is unadvisable:
```
  <body>

    <h1>Site name</h1>

    <header>
      <nav>
        <h2>Site navigation</h2>
      </nav>
    </header>

    <main>
      <h1>Main article heading</h1>
    </main>

    <footer>Site Footer</footer>

  </body>
```

The first h1 in the document should be hoisted to circumvent this poor structure.

**Comment by unor** [comment]
I think this might be a duplicate: https://github.com/validator/validator/issues/167

**Comment by TheDutchCoder** [comment]
Thanks!

**Comment by sideshowbarker** [comment]
> Thanks, I'll guess this will never really get resolved then?

Yeah, the checker is just following the HTML spec requirements here. So the only way this would get resolved otherwise is the outline algorithm in the spec changes. Then I’d update the checker code to match that spec change.
